### PR TITLE
Emagged airlocks can now be repaired + some other airlock code tidying.

### PR DESCRIPTION
**Pull request by Hockaa**
Changes emagging to be identified by the emagged var instead of changing operating to -1.
Emagging simply adds the BROKEN flag to airlocks now.
Tidies up operating var by making it TRUE/FALSE, not 1/0
Using a crowbar on an emagged windoor would spawn electronics and THEN call the shatter proc, which also spawned an electronics board. Both of these would also be given the broken icon state, but still be fully functional and re-usable. I changed the crowbar interaction to no longer rely on the shatter proc and instead refunds the 5 reinforced glass instead of completely shattering it. Both the crowbar interaction and shatter() also now give fried electronics boards.

### CODE REVIEW

**Comment by Hockaa** [comment]
!bugfix

**Comment by Hockaa** [comment]
!review

File: code/game/machinery/doors/airlock.dm [hunk 1]
```diff
@@ -928,7 +928,7 @@ About the new airlock wires panel:
 					src.unlock(TRUE) //force it
 
 /obj/machinery/door/airlock/CanUseTopic(var/mob/user)
-	if(operating < 0) //emagged
+	if(emagged == 1)
```
Comment (mikomyazaki) [inline code/game/machinery/doors/airlock.dm:931]:
```suggestion
	if(emagged)
```

Emagged is binary right? So can just have this.

File: code/game/machinery/doors/airlock_control.dm [hunk 2]
```diff
@@ -22,7 +22,7 @@
 /obj/machinery/door/airlock/proc/execute_current_command()
 	set waitfor = FALSE
 	if(operating)
-		if (operating != -1)	// -1 is emagged.
+		if (emagged == 1)
```
Comment (mikomyazaki) [inline code/game/machinery/doors/airlock_control.dm:25]: Same here.

File: code/game/machinery/doors/door.dm [hunk 3]
```diff
@@ -377,9 +377,10 @@
 /obj/machinery/door/emag_act(var/remaining_charges)
 	if(density && operable())
 		do_animate("spark")
+		emagged = 1
```
Comment (mikomyazaki) [inline code/game/machinery/doors/door.dm:380]:
```suggestion
		emagged = TRUE
```

File: code/game/machinery/doors/windowdoor.dm [hunk 4]
```diff
@@ -120,14 +106,14 @@
...
-		src.operating = 0
+	if(operating == TRUE) //emag again
+		operating = FALSE
 	return 1
 
 /obj/machinery/door/window/close()
-	if (src.operating)
-		return 0
-	src.operating = 1
+	if (src.operating || emagged == 1)
```
Comment (mikomyazaki) [inline code/game/machinery/doors/windowdoor.dm:114]:
```suggestion
	if (src.operating || emagged)
```

File: code/game/machinery/doors/windowdoor.dm [hunk 5]
```diff
@@ -164,16 +150,17 @@
 
 /obj/machinery/door/window/emag_act(var/remaining_charges, var/mob/user)
 	if (density && operable())
-		operating = -1
+		emagged = 1
```
Comment (mikomyazaki) [inline code/game/machinery/doors/windowdoor.dm:153]:
```suggestion
		emagged = TRUE
```

File: code/game/machinery/doors/windowdoor.dm [hunk 6]
```diff
@@ -164,16 +150,17 @@
...
 		sleep(6)
 		open()
+		desc = "A strong door. It keeps trying to close, but is jammed."
 		return 1
 
 /obj/machinery/door/window/attackby(obj/item/I as obj, mob/user as mob)
 
 	//If it's in the process of opening/closing, ignore the click
-	if (src.operating == 1)
+	if (operating == TRUE)
```
Comment (mikomyazaki) [inline code/game/machinery/doors/windowdoor.dm:163]:
```suggestion
	if (operating)
```

File: code/game/machinery/doors/windowdoor.dm [hunk 7]
```diff
@@ -186,40 +173,17 @@
 		return 1
 
 	//If it's emagged, crowbar can pry electronics out.
-	if (src.operating == -1 && I.iscrowbar())
+	if (emagged == 1 && I.iscrowbar())
```
Comment (mikomyazaki) [inline code/game/machinery/doors/windowdoor.dm:176]:
```suggestion
	if (emagged && I.iscrowbar())
```

File: code/game/machinery/doors/airlock.dm (hunk 1 above)
Comment (Hockaa) [reply code/game/machinery/doors/airlock.dm:931]: I checked, and emagged is seen with with numbers like 2 and 0.5, so I assumed I should keep it to 1 and 0. Is that wrong?

File: code/game/machinery/doors/windowdoor.dm [hunk 8]
```diff
@@ -105,12 +91,12 @@
...
 
 /obj/machinery/door/window/open()
-	if (src.operating == 1) //doors can still open when emag-disabled
+	if (operating == TRUE) //doors can still open when emag-disabled
 		return 0
 	if (!ROUND_IS_STARTED)
 		return 0
-	if(!src.operating) //in case of emag
-		src.operating = 1
+	if(!operating) //in case of emag
```
Comment (mikomyazaki) [inline code/game/machinery/doors/windowdoor.dm:98]: This check is unnecessary, if we get to this point we know operating is FALSE already.

**Review by mikomyazaki — CHANGES_REQUESTED** [review]

File: code/game/machinery/doors/door.dm [hunk 9]
```diff
@@ -523,7 +524,7 @@
 			for (var/atom/movable/M in get_turf(src))
 				if (M.density && M != src)
 					addtimer(CALLBACK(src, .proc/autoclose), 60)
```
[the remark this answers is not in the document]
Comment (Hockaa) [reply code/game/machinery/doors/door.dm]: Pretty sure what I added in the push just now is what you were suggesting, but I have a feeling I might have misunderstood something, so I won't resolve it just yet.

[the remark this answers is not in the document]
Comment (mikomyazaki) [reply code/game/machinery/doors/door.dm]: Nah, what you did is one of the ways of resolving it.

**Review by mikomyazaki — APPROVED** [review]
Making a note here that I should refactor everything that treats emagged as not a bool at some point...

**Pull request merged**